signer/src/transaction_signer.rs

```diff
@@ -684,18 +786,144 @@ where
                 tracing::debug!(signature_type = ?request.signature_type, "processing message");
 
                 let db = self.context.get_storage();
-                let accepted_sighash =
-                    Self::validate_bitcoin_sign_request(&db, &request.message).await?;
 
-                let id = accepted_sighash.sighash.into();
+                let id = match msg.id {
+                    WstsMessageId::Dkg(_) => {
+                        tracing::warn!("received message is not allowed in the current context");
+                        return Ok(());
+                    }
+                    WstsMessageId::Sweep(txid) => {
+                        span.record("txid", txid.to_string());
+                        tracing::info!(
+                            signature_type = ?request.signature_type,
+                            "processing message"
+                        );
+
+                        let accepted_sighash =
+                            Self::validate_bitcoin_sign_request(&db, &request.message).await?;
+
+                        accepted_sighash.sighash.into()
+                    }
+                    WstsMessageId::DkgVerification(key) => {
+                        // This is a DKG verification signing round. The data
+                        // provided by the coordinator for signing is expected
+                        // to be the current bitcoin chain tip block hash, which
+                        // we validate and return an error if it does not match
+                        // our view of the current chain tip. We also verify
+                        // that the provided aggregate key matches our latest
+                        // aggregate key.
+
+                        let new_key: PublicKeyXOnly = key.into();
+
+                        // Validate the received message.
+                        Self::validate_dkg_verification_message(
+                            &db,
+                            &new_key,
+                            chain_tip,
+                            &request.message,
+                        )
+                        .await?;
+
+                        tracing::debug!(
+                            signature_type = ?request.signature_type,
+                            "processing message"
+                        );
+
+                        let key = key.into();
```

Comment:
nit: we can reuse `new_key`

Reply:
This disappeared somewhere along the way in #1301